### ISSUE TITLE
Error "Could not get state of profile switch"

### ISSUE THREAD

**Issue by PeteSSMMSS** [mammuth/ha-fritzbox-tools#112]
Hey!
I´m using the new profile switch for a few days now without problems, but today I got a couple of this errors here:

```
2020-09-12 07:53:10 ERROR (MainThread) [custom_components.fritzbox_tools.switch] Could not get state of profile switch
Traceback (most recent call last):
  File "/config/custom_components/fritzbox_tools/switch.py", line 471, in async_update
    status = await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/fritzbox_tools/switch.py", line 472, in <lambda>
    lambda: self.profile_switch.get_state()
  File "/usr/local/lib/python3.8/site-packages/fritzprofiles/fritzprofiles.py", line 74, in get_state
    self.sid = self.login()
  File "/usr/local/lib/python3.8/site-packages/fritzprofiles/fritzprofiles.py", line 31, in login
    sid, challenge = self.get_sid_challenge(self.url + '/login_sid.lua')
  File "/usr/local/lib/python3.8/site-packages/fritzprofiles/fritzprofiles.py", line 24, in get_sid_challenge
    data = lxml.etree.fromstring(r.content)
  File "src/lxml/etree.pyx", line 3237, in lxml.etree.fromstring
  File "src/lxml/parser.pxi", line 1896, in lxml.etree._parseMemoryDocument
  File "src/lxml/parser.pxi", line 1784, in lxml.etree._parseDoc
  File "src/lxml/parser.pxi", line 1141, in lxml.etree._BaseParser._parseDoc
  File "src/lxml/parser.pxi", line 615, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 725, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 654, in lxml.etree._raiseParseError
  File "<string>", line 1
lxml.etree.XMLSyntaxError: Opening and ending tag mismatch: HR line 1 and BODY, line 1, column 186
```

I have a FritzBox 6591 Cable, FRITZ!OS: 07.13 - There are no errors in FritzBox System/Ereignisse Log File so far (By the way is it possible to get the log activities displayed in Home Assistant? So I could trigger notifications when there would be any errors) Thanks again for the work on this custom component, the older version of the profile switch caused definitely more errors

**Comment by AaronDavidSchneider** [comment]
Thanks for reporting! Just to understand correctly: The profile switches work. But you get this errors?

It seems that the html response of the login challenge is broken in your case (e.g. temporary connection error to fritzbox, slow response, etc). In this case home assistant will fail to update the switch. Upon next update of the switch it should get a correct response and get a new login. Is that correct?

The simplest way would be to suppress this error or allow broken html. I will think of something 👍 

Thanks for testing!

**Comment by PeteSSMMSS** [comment]
Yes, that's correct, the switch turned on/off a few times in the night time, but went back to the state it should be. Happened three, four times - No Errors since, and the profile turned on in the morning as expected from an automation trigger.

**Comment by AaronDavidSchneider** [comment]
Do you know if there were some network issues at the same time? e.g. restart of the fritzbox?

**Comment by PeteSSMMSS** [comment]
No network errors that I would know of, and I didn´t restart in the night. Here is a screenshot of the FB Log, the first time the error appears was 5:29 in the morning, so a  bit after the call monitor component logged in - Could that be somehow connected? 
![sc](https://user-images.githubusercontent.com/54634565/92989758-de6bb200-f4d6-11ea-9a20-c40b845a4bea.png)

**Comment by AaronDavidSchneider** [comment]
Can you expand on "4 Meldungen seit 12.09.20 05:09:08" ?
Looks like this could be related?

**Comment by PeteSSMMSS** [comment]
sorry but how do I expand them? When I click on it I get directed to the [help page](https://service.avm.de/help/de/FRITZ-Box-6591-Cable/018/hilfe_syslog_504) - I already set it to Ereignisse = Alle

**Comment by AaronDavidSchneider** [comment]
Ok. Never mind. I will try to find a solution to broken HTML responses and update the code.
Thanks for the report!

**Comment by PeteSSMMSS** [comment]
thank you @AaronDavidSchneider!

**Comment by AaronDavidSchneider** [comment]
I added a proper xml parser that should be able to deal with broken responses. Since I never experienced this error, it would be nice if you could keep an eye on it and notify me, if your error still recurs. Thanks @PeteSSMMSS !

**Comment by mammuth** [comment]
Released as pre-release[ 1.11.2](https://github.com/mammuth/ha-fritzbox-tools/releases/tag/1.11.2)
